KoPrivateGPT/retrieval/base.py

```diff
@@ -18,3 +25,70 @@ def ingest(self, passages: List[Passage]):
     @abstractmethod
     def retrieve_id(self, query: str, top_k: int = 5, *args, **kwargs) -> List[Union[str, UUID]]:
         pass
+
+    def fetch_data(self, ids: List[UUID]) -> List[Passage]:
+        db_origin_list = self.redis_db.get_json(ids)
+        # Sometimes redis doesn't find the id, so we need to filter that db_origin is None.
+        filter_db_origin = list(filter(lambda db_origin: db_origin is not None, db_origin_list))
+        # Check duplicated db instance
+        final_db_origin = self.duplicate_check(filter_db_origin)
+        # fetch data from each db
+        passage_list = self.fetch_each_db(final_db_origin, ids)
+        return passage_list
+
+    def fetch_each_db(self, final_db_origin: dict[tuple, list[int]], ids: List[UUID]):
```

Comment:
Also, I don't know why are you using dict[tuple, list[int]] type. Just use DBOrigin.

Reply:
We came to an agreement through conversation